db/compaction_iterator.cc

```diff
@@ -235,6 +262,11 @@ void CompactionIterator::NextFromInput() {
         }
       }
     } else {
+      if (compaction_listener_)
+        compaction_listener_->OnCompaction(compaction_->level(), ikey_.user_key,
+                                           fromInternalValueType(ikey_.type),
+                                           value_, ikey_.sequence, false);
```

Comment:
Same as explained above, I suggest braces here.

Also, is it possible that we call it on the same key twice, as we've already called it once above?